core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -133,6 +145,50 @@ public Response deleteBucket(@PathParam("bucket") final String bucket) {
   }
 
   /**
+   * @summary gets a bucket and lists all the objects in it
+   * @param bucket the bucket name
+   * @param continuationToken the optional continuation token param
+   * @param maxKeys the optional max keys param
+   * @param prefix the optional prefix param
+   * @return the response object
+   */
+  @GET
+  @Path(BUCKET_PARAM)
+  @ReturnType("ListBucketResult")
+  // TODO(chaomin): consider supporting more request params like prefix and delimiter.
+  public Response getBucket(@PathParam("bucket") final String bucket,
+      @QueryParam("continuation-token") final String continuationToken,
+      @QueryParam("max-keys") final String maxKeys,
+      @QueryParam("prefix") final String prefix) {
+    return S3RestUtils.call(bucket, new S3RestUtils.RestCallable<ListBucketResult>() {
+      @Override
+      public ListBucketResult call() throws S3Exception {
+        Preconditions.checkNotNull(bucket, "required 'bucket' parameter is missing");
+        String bucketPath = AlluxioURI.SEPARATOR + bucket;
+        if (bucketPath.contains(BUCKET_SEPARATOR)) {
+          bucketPath = bucketPath.replace(BUCKET_SEPARATOR, AlluxioURI.SEPARATOR);
+          checkNestedBucketIsUnderMountPoint(bucketPath);
+        }
+
+        checkBucketIsAlluxioDirectory(bucketPath);
```

Comment:
how about having a `String toBucketPath(String bucketName)` method which converts the bucket name to bucket path and validate it? The above code snippet repeats in almost every method.

Reply:
done, see `parseBucketPath()`